src/sentry/utils/redis.py

```diff
@@ -43,11 +42,69 @@ def _shared_pool(**opts):
         return pool
 
 
-def make_rb_cluster(hosts):
-    """Returns a rb cluster that internally shares the pools more
-    intelligetly.
-    """
-    return rb.Cluster(hosts, pool_cls=_shared_pool)
+_make_rb_cluster = functools.partial(rb.Cluster, pool_cls=_shared_pool)
+
+
+def make_rb_cluster(*args, **kwargs):
+    warnings.warn(
+        'Direct Redis cluster construction is deprecated, please use named clusters.',
+        DeprecationWarning,
+    )
+    return _make_rb_cluster(*args, **kwargs)
+
+
+class ClusterManager(object):
+    def __init__(self):
+        self.__clusters = {}
+
+    def get(self, key):
+        cluster = self.__clusters.get(key)
+
+        if cluster is None:
+            # TODO: This would probably be safer with a lock, but I'm not sure
+            # that it's necessary.
+            configuration = options.get('redis.clusters').get(key)
+            if configuration is None:
+                raise KeyError('Invalid cluster name: {}'.format(key))
+
+            cluster = self.__clusters[key] = _make_rb_cluster(**configuration)
+
+        return cluster
+
+
+clusters = ClusterManager()
+
+
+def get_cluster_from_options(backend, options, cluster_manager=clusters, cluster_option_name='cluster', default_cluster_name='default'):
+    cluster_constructor_option_names = frozenset(('hosts',))
```

Comment:
This started as a larger set, but I reduced it to only arguments that were previously supported to encourage switching to named clusters. More detail here: 168c4593d7443e038c92f2591b42ddf2a62674c0